File: C#/PokerProtocol/Commands/Lobby/Career/GetUserCommand.cs
using System;
using System.Collections.Generic;
using System.Text;
using EricUtility;
using PokerWorld.Game;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Lobby.Career
{
    public class GetUserCommand : AbstractLobbyCommand
    {
        protected override string CommandName
        {
            get { return COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "lobbyCAREER_GET_USER";
        
        private readonly string m_Username;

        public string Username
        {
            get { return m_Username; }
        }

        public GetUserCommand(StringTokenizer argsToken)
        {
            m_Username = argsToken.NextToken();
        }

        public GetUserCommand(string p_Username)
        {
            m_Username = p_Username;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_Username);
        }

        public string EncodeResponse(string mail, string display, double money)
        {
            return new GetUserResponse(this, mail, display, money).Encode();
        }
    }
}

File: JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands/lobby/career/GetUserCommand.java
package bluffinmuffin.protocol.commands.lobby.career;

import java.util.StringTokenizer;

import bluffinmuffin.protocol.commands.Command;
import bluffinmuffin.protocol.commands.ICommand;

public class GetUserCommand implements ICommand
{
    public static String COMMAND_NAME = "lobbyCAREER_GET_USER";
    
    private final String m_username;
    
    public GetUserCommand(StringTokenizer argsToken)
    {
        m_username = argsToken.nextToken();
    }
    
    public GetUserCommand(String p_Username)
    {
        m_username = p_Username;
    }
    
    @Override
    public String encodeCommand()
    {
        final StringBuilder sb = new StringBuilder();
        sb.append(GetUserCommand.COMMAND_NAME);
        sb.append(Command.L_DELIMITER);
        sb.append(m_username);
        sb.append(Command.L_DELIMITER);
        return sb.toString();
    }
    
    public String encodeResponse(String mail, String display, double money)
    {
        return new GetUserResponse(m_username, mail, display, money).encodeCommand();
    }
    
    public String getUsername()
    {
        return m_username;
    }
}
